lib/errors/index.ts

```diff
@@ -90,7 +95,22 @@ export class ArsenalError extends Error {
     customizeDescription(description: string): ArsenalError {
         const type = this.#type;
         const code = this.#code;
-        return new ArsenalError(type, code, description);
+        const metadata = this.#metadata;
+        const err = new ArsenalError(type, code, description, metadata);
```

Comment:
Because Maps are pointers and are subject to state modifications, it would be a huge improvement to duplicate the `Map` structure in the new error, just like we are duplicating the error. Pure code tends to avoid bugs. In fact, doing something like this:

```javascript
const foo = () => {
  const error = errors.InternalError
  const newError = error.customizeDescription('bla')
  newError.addMetadataEntry('bar', 'baz')
  error.medatata.get('bar') // ==> 'baz'
  // We probably want error to not be touched.
}
```

We could probably solve the problem by just doing `const metadata = new Map(this.#metadata)`.